n3fit/src/n3fit/backends/keras_backend/MetaModel.py

```diff
@@ -414,6 +408,17 @@ def load_identical_replicas(self, model_file):
         for i_replica in range(self.num_replicas):
             self.set_replica_weights(weights, i_replica)
 
+    def save_weights(self, file, save_format="h5"):
+        """
+        Compatibility function for tf < 2.16
+        """
+        try:
+            super().save_weights(file, save_format=save_format)
+        except TypeError:
+            new_file = file.with_suffix(".weights.h5")
+            super().save_weights(new_file)
+            shutil.move(new_file, file)
```

Comment:
Maybe add a comment here which branch corresponds to which version. Also not sure what happens here, the second branch requires a particular suffix? Or doesn't allow you to overwrite an existing file?

Reply:
I'll add it as a comment but, the reason is that for keras 3 `save_format` doesn't exist anymore and instead the format is read directly from the extension [docs](https://keras.io/api/models/model_saving_apis/weights_saving_and_loading/#save_weights-method).

So you are forced to save it to a `.weights.h5` file.